### PR TITLE
[language-platform] adding route to check for SCIP availability

### LINKED ISSUE #45078: RFC 740: Adjust src-cli to upload SCIP files
Currently `src codeintel upload` will translate a SCIP file to LSIF prior to upload. It should instead check with the instance to see if it can handle SCIP uploads. This can be done with a `HEAD` request to the upload endpoint the which will return a new `Accept` header, preferred; or by version check which will be less future-proof.

Keep the current behavior in place for instances that do not accept SCIP. For instances that do accept SCIP, we need to upload SCIP files as-is, and translate LSIF files to SCIP. @efritz can expose the utility to do this translation as we've already written a working/tested version for oobmigrations on _slightly_ different data.

Follow-up concerns that should be raised and new tickets filed:

- [x] We should only upload SCIP if the instance accepts it (so that we don't break oobmigration concerns).
- [x] There's a few places in src-cli that we try to read the metadata field from the LSIF file. This should be updated to read from the SCIP index as well.

### PR DESCRIPTION
**Pull request by Numbers88s**
Fixes #45078 

Adjust src-cli to upload SCIP files by adding route to check for SCIP availability and modify response depending on availability.

## Test plan
Pass all the test.

### CODE REVIEW

File: enterprise/internal/codeintel/uploads/transport/http/handler.go [hunk 1]
```diff
@@ -63,6 +63,12 @@ func newHandler(
 	return handler
 }
 
+func IsSCIPAvailableHandler() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	})
+}
+
```
Comment (efritz) [inline enterprise/internal/codeintel/uploads/transport/http/handler.go:71]:
```suggestion
var noopHandler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
	w.WriteHeader(http.StatusOK)
})
```

Could make it even easier/more generic.

File: cmd/frontend/internal/httpapi/httpapi.go [hunk 2]
```diff
@@ -47,8 +47,10 @@ type Handlers struct {
 	BatchesChangesFileGetHandler    http.Handler
 	BatchesChangesFileExistsHandler http.Handler
 	BatchesChangesFileUploadHandler http.Handler
-	NewCodeIntelUploadHandler       enterprise.NewCodeIntelUploadHandler
 	NewComputeStreamHandler         enterprise.NewComputeStreamHandler
+
+	NewCodeIntelUploadScipAvailableHandler http.Handler
```
Comment (efritz) [inline cmd/frontend/internal/httpapi/httpapi.go:52]: Not sure we need to thread this through along with the other handler. If we do https://github.com/sourcegraph/sourcegraph/pull/45269/files?diff=split&w=1#r1041355534, we could simply just install that simple-200 handler on the route whether or not the other one is enabled. It will have the same effect to the client.

File: enterprise/internal/codeintel/uploads/transport/http/handler.go (hunk 1 above)
Comment (Numbers88s) [reply enterprise/internal/codeintel/uploads/transport/http/handler.go]: Done.

File: cmd/frontend/internal/httpapi/httpapi.go (hunk 2 above)
Comment (Numbers88s) [reply cmd/frontend/internal/httpapi/httpapi.go]: Done, de-threaded and now we use the noopHandler.

File: cmd/frontend/internal/httpapi/httpapi.go [hunk 3]
```diff
@@ -47,8 +47,10 @@ type Handlers struct {
 	BatchesChangesFileGetHandler    http.Handler
 	BatchesChangesFileExistsHandler http.Handler
 	BatchesChangesFileUploadHandler http.Handler
-	NewCodeIntelUploadHandler       enterprise.NewCodeIntelUploadHandler
 	NewComputeStreamHandler         enterprise.NewComputeStreamHandler
+
+	NewCodeIntelUploadScipAvailableHandler http.HandlerFunc
```
Comment (efritz) [inline cmd/frontend/internal/httpapi/httpapi.go:52]: Why is this here but not in the internal routes?

File: cmd/frontend/internal/httpapi/httpapi.go [hunk 4]
```diff
@@ -210,6 +215,8 @@ func NewInternalHandler(
 	m.Get(apirouter.ComputeStream).Handler(trace.Route(newComputeStreamHandler()))
 
 	m.Get(apirouter.LSIFUpload).Handler(trace.Route(newCodeIntelUploadHandler(false)))
+	m.Get(apirouter.SCIPUpload).Handler(trace.Route(newCodeIntelUploadHandler(false)))
+	m.Get(apirouter.SCIPUploadExists).Handler(trace.Route(noopHandler))
```
Comment (Numbers88s) [inline cmd/frontend/internal/httpapi/httpapi.go:219]: It is in internal, unless I am missing what you are referring to.

File: cmd/frontend/internal/httpapi/httpapi.go (hunk 3 above)
Comment (Numbers88s) [reply cmd/frontend/internal/httpapi/httpapi.go:52]: Is this what you are referring to https://github.com/sourcegraph/sourcegraph/pull/45269/files/7b49683751172f3983461e1d2052b31fa0fe0d85#r1042604099

Comment (efritz) [reply cmd/frontend/internal/httpapi/httpapi.go:52]: Why do you still have a field here?

Comment (Numbers88s) [reply cmd/frontend/internal/httpapi/httpapi.go:52]: I was threading it to keep everything in handlers, but regardless moved it to the file so no more threading it.

**Review by efritz — APPROVED** [review]
